shared/src/protocol/navigation/location.rs

```diff
@@ -0,0 +1,82 @@
+use std::fmt::Display;
+
+use euclid::{approxeq::ApproxEq, Point2D};
+use num_traits::FromPrimitive;
+use serde::{Deserialize, Serialize};
+
+use crate::solar_systems::{self, Solar};
+
+#[derive(Debug, Copy, Hash, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct SolarSystemId(pub i64);
+
+impl SolarSystemId {
+    #[must_use]
+    pub fn into_location<T: FromPrimitive>(self) -> Option<T> {
+        T::from_i64(self.0)
+    }
+}
+
+#[cfg(feature = "redis")]
+mod redis {
+    use redis::{FromRedisValue, ToRedisArgs};
+
+    use super::SolarSystemId;
+
+    impl FromRedisValue for SolarSystemId {
+        fn from_redis_value(v: &redis::Value) -> redis::RedisResult<Self> {
+            let value = i64::from_redis_value(v)?;
+            Ok(Self(value))
+        }
+    }
+
+    impl ToRedisArgs for SolarSystemId {
+        fn write_redis_args<W>(&self, out: &mut W)
+        where
+            W: ?Sized + redis::RedisWrite,
+        {
+            self.0.write_redis_args(out)
+        }
+    }
+}
+
+impl Display for SolarSystemId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Pilot {
```

Comment:
I'd rather this one be called Universe than Pilot. `location::Universe` is more accurate of what it is, since it's a way to identify a location within the universe, not just a single system.

Reply:
Done.